uiowa/uiowa
# Uninstall `node_view_permissions` mode, add `private_content` module to SPPA split

- Resolves #3925 
- Adds an update hook that attempts to pull in all the updated config related to roles and then rebuilds node access with the hopes of preventing any access denied messages being shown to users.

# How to test
Compare file changes against https://github.com/uiowa/uiowa/pull/3833 to see that all the `view any...` and `view own...` content type permissions have been removed. This excludes `view any unpublished` or `view any webform submissions`.

## Default
`blt ds`

### Stories
- As an editor, publisher, and webmaster, I can create and edit article, page, and person content.
- As an anonymous user, I am able to view published article, page, and person content.

## Admissions
`blt ds --site brand.uiowa.edu`

### Stories
- As an editor, publisher, and webmaster, I can create and edit area of study, event, scholarship, student profile, and transfer tips content.
- As an anonymous user, I am able to view published area of study, event, scholarship, student profile, and transfer tips content.

## Brand
`blt ds --site brand.uiowa.edu`

### Stories
- As a collaborator, I am able to create lockup content.
- As a collaborator, I am able to edit my own lockup content.
- As an editor, I am able to move lockup content from "Draft" to "Needs review."
- As a publisher, and webmaster, I am able to approve and publish lockup content.

## Grad
`blt ds --site grad.uiowa.edu`

### Stories
- As an editor, publisher, and webmaster, I am able to create and edit mentor, scholar, scholarship (fellowship), student profile, and thesis defense content.
- As an anonymous user, I am able to view published mentor, scholar, scholarship, student profile, and thesis defense content.

## HR
`blt ds --site hr.uiowa.edu`

### Stories
- As an editor, publisher, and webmaster, I am able to create and edit FAQ content.
- As an anonymous user, I am able to view published FAQ content.

## IGHN
`blt ds --site ighn.international.uiowa.edu`

### Stories
- As an editor, publisher, and webmaster, I am able to create and edit ighn person content.
- As an anonymous user, I am able to view published ighn person content.

## CLAS Policy
`blt ds --site policy.clas.uiowa.edu`

### Stories
- As an editor, publisher, and webmaster, I am able to create and edit book content.
- As an anonymous user, I am able to view published book content.

## SPPA
`blt ds --site sppa.uiowa.edu`

### Stories
- As a site owner/editor, I can choose to make a page "private" (under Promotion options) and it can only be seen by editors, publishers, students, and webmasters.
- As an anonymous, authenticated, requester, and viewer, I am not able to view opportunity content or https://sppa.local.drupal.uiowa.edu/opportunities
